engine/src/group.cpp

```diff
@@ -1031,6 +1021,16 @@ void MCGroup::applyrect(const MCRectangle &nrect)
 
 void MCGroup::removereferences()
 {
+    if (this == MCmenubar)
```

Comment:
@livecodealex Could you check `MCmenubar.IsValid()` here first just to make sure. So:

```
if (!MCmenubar.IsValid() || this == MCmenubar)
```
Same below... shouldn't ever be invalid here but if it was it would crash.